compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -16977,6 +16976,9 @@ private STNode parseStatementStartsWithOpenBracket(STNode annots, boolean isRoot
                 case TUPLE_TYPE_DESC:
                     // If the member type was figured out as a tuple-type-desc member, then parse the
                     // remaining members as tuple type members and be done with it.
+                    member = createMemberOrRestNode(STNodeFactory.createEmptyNodeList(), member);
+                    return parseAsTupleTypeDesc(annots, openBracket, memberList, member, isRoot);
```

Comment:
For samples like bellow
```bal
public function main() {
    [T...] a;
}
``` 
This will create a member descriptor containing rest descriptor inside

Reply:
I think we should return the member descriptor from `parseStatementStartBracketedListMember` itself instead of returning just the type desc and creating the member descriptor here

Reply:
When we create the member descriptor here since both tuple type desc and typed binding patterns are parsed in `parseStatementStartBracketedListMember` in a case like `[[int, int], int] [[a,b], c]`, `[a,b]` will be parsed into a `STMemberTypeDescriptorNode`. 
We can either  create member descriptor outside the function (as earlier), use separate functions or change STNode type later in the parser.

Reply:
I see. We rearrange the node in the binding pattern scenario anyway.
e.g. a `SimpleNameReferenceNode` will be converted to a `CaptureBindingPatternNode`. 

Therefore shall add the `MemberTypeDescriptorNode` conversion at the same place?

https://github.com/ballerina-platform/ballerina-lang/blob/31e1d2c7b78e23ab880f16650ac74a11dcc0c89b/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L18650

Reply:
Is there a problem with creating the member descriptor here and having a separate match case for `REST_TYPE`, otherwise it will add a lot of somewhat duplicate code?

Reply:
Reverted as discussed.